src/connection.rs

```diff
@@ -1,47 +1,40 @@
-use std::sync::mpsc;
+/*use std::sync::mpsc;
 #[cfg(feature="voice")]
 use std::collections::HashMap;
 
-use websocket::client::{Client, Sender, Receiver};
-use websocket::stream::WebSocketStream;
-
 use serde_json;
 
 use model::*;
-use internal::Status;
-#[cfg(feature="voice")]
-use voice::VoiceConnection;
+//#[cfg(feature="voice")]
+//use voice::VoiceConnection;
 use {Result, Error, SenderExt, ReceiverExt};
+use tokio_core::reactor::Core;
+use async::single_conn::{self, SingleConnection};
 
 const GATEWAY_VERSION: u64 = 6;
 
 #[cfg(feature="voice")]
 macro_rules! finish_connection {
-	($($name1:ident: $val1:expr),*; $($name2:ident: $val2:expr,)*) => { Connection {
+	($($name1:ident: $val1:expr),*; $($name2:ident: $val2:expr),*) => { Connection {
 		$($name1: $val1,)*
 		$($name2: $val2,)*
 	}}
 }
 #[cfg(not(feature="voice"))]
 macro_rules! finish_connection {
-	($($name1:ident: $val1:expr),*; $($name2:ident: $val2:expr,)*) => { Connection {
+	($($name1:ident: $val1:expr),*; $($name2:ident: $val2:expr),*) => { Connection {
 		$($name1: $val1,)*
 	}}
 }
 
 /// Websocket connection to the Discord servers.
 pub struct Connection {
```

Comment:
in my work, i pushed the core to its own separate thread and pushed futures into it. i used regular std mpsc queues to handle constructing futures and allowed a cleaner seperation of sync and async code. but i don't know if that way of structuring things would even work all that well